packages/provider/src/init.ts

```diff
@@ -84,58 +84,31 @@ export const initWallet = (projectAccessKey: string, partialConfig?: Partial<Pro
     }
   }
 
-  const rpcProviders: Record<number, ethers.JsonRpcProvider> = {}
-
-  // Find any new networks that aren't already defined in sequence.js
-  // and add them to the list of networks, (they must have a rpcUrl and chainId)
-  const newNetworks = (config.networks?.filter(n => {
-    // eslint-disable-next-line
-    n.rpcUrl !== undefined && n.chainId !== undefined && !allNetworks.find(an => an.chainId === n.chainId)
-  }) ?? []) as NetworkConfig[]
-
-  // Override any information about the networks using the config
-  const combinedNetworks = allNetworks
-    .map(n => {
-      const network = config.networks?.find(cn => cn.chainId === n.chainId)
-      return network ? { ...n, ...network } : n
-    })
-    .concat(newNetworks)
-    .map(network => {
-      // don't double-append in the case the user has already included their access key in the rpc URL
+  let networks: NetworkConfig[] = []
+
+  const updateNetworks = (connectedNetworks: NetworkConfig[] = []) => {
+    networks = mergeNetworks(allNetworks, connectedNetworks, config.networks ?? [])
+
+    // Append projectAccessKey to network rpcUrls
+    networks = networks.map(network => {
+      // Don't double-append in the case the user has already included their access key in the rpc URL
       if (network.rpcUrl.includes(projectAccessKey)) {
         return network
       }
 
-      // this will probably break non-sequence RPC provider URLs.
+      // XXX: This will probably break non-sequence RPC provider URLs.
```

Comment:
lets just search for "sequence.app" in the url then, and skip adding this if thats the case..?